staging/src/k8s.io/apimachinery/pkg/util/net/interface.go

```diff
@@ -149,90 +163,101 @@ func getIPFromInterface(intfName string, nw networkInterfacer) (net.IP, error) {
 	return nil, nil
 }
 
-func flagsSet(flags net.Flags, test net.Flags) bool {
-	return flags&test != 0
-}
-
-func flagsClear(flags net.Flags, test net.Flags) bool {
-	return flags&test == 0
+// memberOF tells if the IP is of the desired family. Used for checking interface addresses.
+func memberOf(ip net.IP, family AddressFamily) bool {
+	ipFamily := familyIPv4
+	if ip.To4() == nil {
+		ipFamily = familyIPv6
+	}
+	return ipFamily == family
 }
 
-func chooseHostInterfaceNativeGo() (net.IP, error) {
-	intfs, err := net.Interfaces()
+// chooseIPFromHostInterfaces looks at all system interfaces, trying to find one that is up that
```

Comment:
There's not one RFC that defines this. See https://golang.org/src/net/ip.go for the definition of the IsGlobalUnicast() function from the net package, which is used.  Seems to be, that referring someone to four RFCs, and then having them weed through them to understand the definition, is not going to be useful. Alternatively, I could indicate that it is a non-LB, non-link local, non-P2P address.  Will try rewording.